lib/FlyBy.pm

```diff
@@ -255,7 +255,11 @@ sub all_keys {
 
 sub values_for_key {
     my ($self, $key) = @_;
-    return (sort { $a cmp $b } keys %{$self->index_sets->{$key}});
+
+    my $set = $self->index_sets->{$key};
+    my @values = grep { my $v = $set->{$_}; $v && %{$v->{elements}} } keys %$set;
```

Comment:
Maybe I didn't get the entire logic right. I did this:

```
--- a/lib/FlyBy.pm
+++ b/lib/FlyBy.pm
@@ -97,7 +97,7 @@ sub _from_index {
         $value = substr($value, 1);
     }
 
-    if (not $add_missing_key and not exists $index_sets->{$key}) {
+    if (not $add_missing_key) {
         $result = ($negated) ? $self->_full_set : $self->_full_set->empty_clone;    # Avoiding auto-viv on request
     } else {
         $index_sets->{$key}{$value} //= $self->_full_set->empty_clone;              # Sets which do not (yet) exist in the index are null.
```
Basically when the `$add_missing_key is set but the key does not exist`, it will create the empty clone for the key. But it fails a lot of tests. Maybe you have an idea of what to do if that `elsif`